proto/cosmos/slashing/genesis.proto

```diff
@@ -6,44 +6,54 @@ option go_package            = "github.com/cosmos/cosmos-sdk/x/slashing/types";
 import "gogoproto/gogo.proto";
 import "cosmos/slashing/slashing.proto";
 
-// GenesisState - all slashing state that must be provided at genesis
+// GenesisState defines all slashing state that must be provided at genesis.
 message GenesisState {
+  // params defines all the paramaters of related to deposit.
   Params params = 1 [
-    (gogoproto.casttype) = "Params",
     (gogoproto.nullable) = false
   ];
 
+  // signing_infos represents a map between validator addresses and their
+  // signing infos.
   repeated SigningInfo signing_infos = 2 [
     (gogoproto.moretags) = "yaml:\"signing_infos\"",
     (gogoproto.nullable) = false
   ];
 
+  // signing_infos represents a map between validator addresses and their
+  // missed blocks.
   repeated ValidatorMissedBlocks missed_blocks = 3 [
     (gogoproto.moretags) = "yaml:\"missed_blocks\"",
     (gogoproto.nullable) = false
   ];
 }
 
-// SigningInfo stores validator signing info of corresponding address
+// SigningInfo stores validator signing info of corresponding address.
 message SigningInfo {
+  // address is the validator address.
   string address = 1;
```

Comment:
We used to use `map[string]SigningInfo`, hence the string here.

Any objections to change this to `bytes addresses = 1 [(gogoproto.casttype) = "ValAddress"];`

Reply:
Yes. This removes the human-readable aspect from the genesis file (with respect to this state).